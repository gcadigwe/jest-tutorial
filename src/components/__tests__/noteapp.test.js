import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import NoteApp from "../Noteapp";

afterEach(() => {
  cleanup();
});

test("should render non-completed note", () => {
  const noteItem = {
    id: 1,
    title: "Demo",
    body: "Why i love javascript",
    completed: false,
  };

  render(<NoteApp note={noteItem} />);
  const noteAppElement = screen.getByTestId("note-1");
  expect(noteAppElement).toHaveTextContent("Demo");
  expect(noteAppElement).toHaveClass("not-completed");
});

test("should render completed note", () => {
  const noteItem = {
    id: 2,
    title: "Demo 2",
    body: "Why i don't like javascript",
    completed: true,
  };

  render(<NoteApp note={noteItem} />);
  const noteAppElement = screen.getByTestId("note-2");
  expect(noteAppElement).toHaveTextContent("Demo 2");
  expect(noteAppElement).toHaveClass("completed");
});

// snapshot test

test("matches snapshot", () => {
  const noteItem = {
    id: 1,
    title: "Demo",
    body: "Why i love javascript",
    completed: false,
  };

  const tree = renderer.create(<NoteApp note={noteItem} />).toJSON();
  expect(tree).toMatchSnapshot();
});
