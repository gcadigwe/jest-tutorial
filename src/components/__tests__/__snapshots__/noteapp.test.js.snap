// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="not-completed"
  data-testid="note-1"
>
  <h1>
    Demo
  </h1>
</div>
`;
